src/WebJobs.Extensions.DurableTask.Analyzers/CodefixProviders/Orchestrator/TimerCodeFixProvider.cs

```diff
@@ -118,50 +135,82 @@ private bool TryGetCancellationTokenParameter(SyntaxNode expression, SemanticMod
+                            parameter = node.First().ToString();
                             return true;
                         }
                     }
                 }
             }
-            else
+
+            parameter = null;
+            return false;
+        }
+
+        private bool TryGetObjectCreationExpressionChildNodes(SyntaxNode argument, out IEnumerable<SyntaxNode> nodes) => TryGetChildNodes(argument, SyntaxKind.ObjectCreationExpression, out nodes);
+
+        private bool TryGetSimpleMemberAccessExpressionChildNodes(SyntaxNode argument, out IEnumerable<SyntaxNode> nodes) => TryGetChildNodes(argument, SyntaxKind.SimpleMemberAccessExpression, out nodes);
+
+        private bool TryGetIdentifierNameChildNodes(SyntaxNode argument, out IEnumerable<SyntaxNode> nodes) => TryGetChildNodes(argument, SyntaxKind.IdentifierName, out nodes);
+
+        private bool TryGetChildNodes(SyntaxNode argument, SyntaxKind kind, out IEnumerable<SyntaxNode> nodes)
+        {
+            if (argument == null)
             {
-                var identifierNameEnumerable = argument.ChildNodes().Where(x => x.IsKind(SyntaxKind.IdentifierName));
-                if (identifierNameEnumerable.Any())
-                {
-                    var identifierName = identifierNameEnumerable.First();
-                    var typeInfo = semanticModel.GetTypeInfo(identifierName);
-                    if (typeInfo.Type != null)
-                    {
-                        var type = typeInfo.Type.ToString();
-                        if (string.Equals(type, typeToCompare))
-                        {
-                            typeReference = identifierNameEnumerable.First().ToString();
-                            return true;
-                        }
-                    }
-                }
+                nodes = null;
+                return false;
+            }
+
+            if (argument.IsKind(kind))
+            {
+                nodes = new List<SyntaxNode>() { argument };
+                return true;
+            }
+            
+            nodes = argument.ChildNodes().Where(x => x.IsKind(kind));
+            
+            if (nodes.FirstOrDefault() != null)
+            {
+                return true;
             }
 
-            typeReference = null;
             return false;
         }
 
+        private bool TryGetTypeName(SemanticModel semanticModel, SyntaxNode identifier, out string typeName)
+        {
+            var typeInfo = semanticModel.GetTypeInfo(identifier);
+            if (typeInfo.Type != null)
+            {
+                typeName = typeInfo.Type.ToString();
+                return true;
+            }
+
+            typeName = null;
+            return false;
+        }
+        
+        private string GetTypeWithoutNamespace(string type)
+        {
+            var splitString = type.Split('.');
+            var length = splitString.Length;
+            return splitString[length - 1];
+        }
```

Comment:
This works, but it's more efficient to use `type.LastIndexOf('.')` instead of `type.Split('.')`. String splitting is generally more expensive because it allocates multiple string objects in memory. I've even seen cases where this has resulted in massive performance problems because doing it too frequently can cause the .NET Garbage Collector to get really busy. That's probably not going to be an issue here, but preferring `string.IndexOf` and `string.LastIndexOf` instead of `string.Split` is a good habit to get into.